examples/cookbook.cpp

```diff
@@ -360,15 +360,11 @@ static bool ThrowError(JSContext* cx, JS::HandleObject global,
   JS::RootedString filenameStr(cx, JS_NewStringCopyZ(cx, filename));
   if (!filenameStr) return false;
 
-  JS::RootedValueArray<3> args(cx);
-  args[0].setString(messageStr);
-  args[1].setString(filenameStr);
-  args[2].setInt32(lineno);
-  JS::RootedValue exc(cx);
-  // The JSAPI code here is actually simulating `throw Error(message)` without
-  // the new, as new is a bit harder to simulate using the JSAPI. In this case,
-  // unless the script has redefined Error, it amounts to the same thing.
-  if (!JS_CallFunctionName(cx, global, "Error", args, &exc)) return false;
+  JS::Rooted<JS::Value> exc(cx);
```

Comment:
We probably want to show `JS::CaptureCurrentStack` as well.

Reply:
I don't really understand how that works, since it seems like `JS_SetPendingException` by default does *some* sort of stack-capturing. But I don't know how that works, since the resulting exception seems to have an empty stack property.

Anyway, you know this better, so I just did what you said and added in `JS::CaptureCurrentStack`, which seems to work.